File: Algorithm/Programmers/Level1/같은 숫자는 싫어.js
function solution(arr){
  let answer = []; // 담아서 리턴할 빈 배열
  
  for(let i = 0; i < arr.length; i++) {
      if(arr[i] !== arr[i+1]) answer.push(arr[i]);
  }
  
  return answer;
}

File: Algorithm/Programmers/Level1/같은 숫자는 싫어.java
// https://school.programmers.co.kr/learn/courses/30/lessons/12906?language=java

import java.util.*;

class Solution {
    public int[] solution(int []arr) {
        List<Integer> list = new ArrayList<>();
        list.add(arr[0]);
        
        for (int i = 1; i < arr.length; i++) {
            if (arr[i-1] != arr[i]) list.add(arr[i]);
        }
        
        int[] answer = list.stream().mapToInt(Integer::intValue).toArray();
        
        return answer;
    }
}